include/common/serial_interface.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+// Posix includes
+#include <fcntl.h>
+#include <unistd.h>
+
+// Standard C includes
+#include <cstdint>
+
+// Standard C++ includes
+#include <stdexcept>
+
+namespace BoBRobotics {
+//----------------------------------------------------------------------------
+// SerialInterface
+//----------------------------------------------------------------------------
+class SerialInterface
```

Comment:
This class looks like it could be used as a replacement for ``SerialReader`` in the GPS code - maybe one to bug and do later